RNA-seq/01-qc_trim_quant.md

```diff
@@ -0,0 +1,244 @@
+
+Trimming reads may result in short reads, which may effect gene expression estimates ([Williams et al. _BMC Bioinformatics._ 2016.](https://doi.org/10.1186/s12859-016-0956-2)).
+Using `--length_required 20` means that reads shorter than 20bp will be discarded (similar to what was used in Srivastava et al. above).
+
+#### `--report_title`
+
+When we look at the HTML report, it's helpful to quickly identify what sample the report is for. Using `--report title "SRR585570"` means that the report will be titled "SRR585570" rather than the default ("fastp report").
+
+#### `--json` and `--html`
+
+With these options, we're specifying where the [JSON](https://en.wikipedia.org/wiki/JSON) and HTML reports will be saved (in the `QC/gastric_cancer/fastp_output/` we created) and what the filenames will be. 
+Including the sample name in the filenames again may help us with project organization.
+
+**If we look at** `QC/gastric_cancer/fastp_output/SRR585570_fastp.json` **or the top of the HTML report, we can see that fastp reports certain metrics before and after filtering, which can be very useful in making analysis decisions.**
+
+### Quantification with Salmon
+
+We'll use [Salmon](https://combine-lab.github.io/salmon/) for quantifying transcript expression ([documentation](http://salmon.readthedocs.io/en/latest/)).
+Salmon ([Patro, et al. _Nature Methods._ 2017.](https://doi.org/10.1038/nmeth.4197)) is fast and requires very little memory, which makes it a great choice for running on your laptop during training.
+We can use the output for downstream analyses like differential expression analysis and clustering.
+We use Salmon in mapping mode, with mapping validation enabled, using the following command:
+
+```bash
+# We perform quantification on the files that have been trimmed
+# and use the index generated with -k 23, as this may "improve sensitivity"
+# per the Salmon documentation
+salmon quant -i index/Homo_sapiens/short_index \
+	-l A \
+	-1 data/fastq/gastric_cancer/SRR585570/SRR585570_fastp_1.fastq.gz \
+	-2 data/fastq/gastric_cancer/SRR585570/SRR585570_fastp_2.fastq.gz \
+	-o data/quant/gastric_cancer/SRR585570 \
+	--validateMappings --rangeFactorization 4 \
+	--gcBias --seqBias
+```
+
+Below, we'll walk through the arguments/options we used to run `salmon quant`.
+
+#### Transcriptome index: `-i`
+
+Salmon requires a set of transcripts (what we want to quantify) in the form of a transcriptome index built with `salmon index`.
+Building an index can take a while (but you only have to do it once!), so we've built the one we use today ahead of time.
+Before we use it, we'll take a moment to give a bit of background.
+
+You can see how we obtained this index and others [on GitHub](https://github.com/AlexsLemonade/training-txome-prep).
+Note that we used Homo sapiens GRCh38, Ensembl release 95.
+It is important to keep track of what build, resource, and files were used and putting our shell scripts on GitHub allows us to do that.  
```

Comment:
```suggestion
For reproducibility purposes, it's important to keep track of what build, resource, and files were used and putting our shell scripts on GitHub allows us to do that.  
```

Reply:
I think it is important for broader, i.e., "getting things to work together," purposes.